### PR TITLE
CO-127: harden run-review direct-exec symlink handling

### PR DESCRIPTION
**Pull request by Kbediako**
## What
- harden `scripts/run-review.ts` direct-exec detection to accept both resolved and realpath entry URLs under `NODE_OPTIONS=--preserve-symlinks-main`
- add focused `tests/run-review.spec.ts` coverage for same-directory preserved-symlink direct execution while keeping the existing pipe-close subprocess regression intact
- register the CO-127 docs-first packet and refresh the task/workpad mirrors for the rework attempt

## Why
- current `main` still has no installed standalone `run-review` bin, but same-directory symlinked direct execution of the helper was a real silent exit-0/no-output seam on the existing source/dist helper surfaces
- CO-127 is the bounded follow-up to close that seam without reopening CO-114 timing work or weakening `does not crash when stdout pipe closes early`

## Validation
- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npm run build`
- `npm run lint`
- `npm run test`
- `npm run docs:check`
- `npm run docs:freshness`
- `npm run repo:stewardship`
- `node scripts/diff-budget.mjs`
- `npm run pack:smoke`
- docs-review child stream: `.runs/linear-b3286a9a-9cef-45a5-bd8a-532856a1188d-docs-review-rework/cli/2026-04-11T22-05-57-080Z-760b999c/review/telemetry.json` (`review_outcome=clean-success`)
- standalone review: `.runs/linear-b3286a9a-9cef-45a5-bd8a-532856a1188d/cli/2026-04-11T14-43-18-011Z-10ef4027/review/telemetry.json` (`review_outcome=clean-success`)
- explicit elegance review: `out/linear-b3286a9a-9cef-45a5-bd8a-532856a1188d/manual/20260412T081636Z-elegance-review.md`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved run-review direct-execution detection to handle symlinked and resolved entry forms for more reliable invocation.

* **Tests**
  * Added focused tests covering direct-exec and same-directory symlink scenarios, including help-output and stdout-pipe edge-case regression checks.

* **Documentation**
  * Added PRD, tech spec, action plan and task checklists detailing scope, validation gates and handoff steps.

* **Chores**
  * Updated task registry and docs freshness metadata to register the new work items.

### CODE REVIEW

File: tests/run-review.spec.ts [hunk 1]
```diff
@@ -1703,6 +1704,61 @@ describe('shouldUseFreshDist', () => {
...
+      dirname(directEntry),
+      `run-review-link-${process.pid}-${Date.now()}${directEntry.endsWith('.ts') ? '.ts' : '.js'}`
+    );
+
+    try {
+      await symlink(basename(directEntry), linkPath);
+
+      const directArgs = directEntry.endsWith('.ts')
+        ? ['--loader', 'ts-node/esm', linkPath, '--help']
+        : [linkPath, '--help'];
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply tests/run-review.spec.ts:1739]: Addressed in commit `eb074aa55`: this subprocess proof now keeps the freshness-based source/dist selection and adds an explicit 15s timeout so source-checkout fallback stays truthful while the CI run no longer flakes on `ts-node/esm` cold start.

File: docs/PRD-linear-b3286a9a-9cef-45a5-bd8a-532856a1188d.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+## Traceability
+- Linear issue: `CO-127` / `b3286a9a-9cef-45a5-bd8a-532856a1188d`
+- Linear URL: https://linear.app/asabeko/issue/CO-127/co-harden-run-review-direct-exec-symlink-handling-without-regressing
+- Source issue: `CO-114` / `54387f04-30aa-436a-9901-690c0e9cfcee`
+- Prior rework PR: `#402` / `https://github.com/Kbediako/CO/pull/402` (closed on 2026-04-11 before merge)
+
+## Summary
+- Problem Statement: `scripts/run-review.ts` still used a stricter direct-execution guard than `bin/codex-orchestrator.ts` under `NODE_OPTIONS=--preserve-symlinks-main`, and current `main` reproduced a real silent no-op when the helper was invoked through a same-directory symlink. The product surface is still narrower than the CLI bin surface because there is no installed standalone `run-review` bin, but the helper direct-exec seam itself was still worth hardening.
+- Desired Outcome: keep the product-surface docs truthful about the absence of an installed `run-review` bin while landing the bounded direct-exec hardening in `scripts/run-review.ts` and preserving the subprocess harness.
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply docs/PRD-linear-b3286a9a-9cef-45a5-bd8a-532856a1188d.md]: Addressed in commit `eb074aa55`: added the comma in the PRD sentence for readability.

File: docs/TECH_SPEC-linear-b3286a9a-9cef-45a5-bd8a-532856a1188d.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+- Delegated surface planned after packet draft:
+  - audited `docs-review` child stream for the refreshed packet, completed with clean review telemetry before handoff
+- No same-issue child lane during the reset turn:
+  - the Rework reset requires parent-owned branch/workpad mutation first, so the turn-level parallelization decision is `forbid_parallel` / `parent_only_mutation`
+
+## Design
+- Prove the actual supported repo-local and packaged entrypoints from `package.json`, launcher code, docs, and direct runtime behavior under `NODE_OPTIONS=--preserve-symlinks-main`.
+- Keep the product-surface docs truthful: there is still no installed standalone `run-review` bin.
+- Harden `scripts/run-review.ts` `isDirectExecution(...)` to accept both resolved and realpath entry URLs, which fixes the reproduced same-directory symlink direct-exec gap for the existing source/dist helper surfaces.
+- Keep the existing `does not crash when stdout pipe closes early` regression intact and use focused coverage for the direct-exec seam instead of widening heavy subprocess coverage unnecessarily.
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply docs/TECH_SPEC-linear-b3286a9a-9cef-45a5-bd8a-532856a1188d.md]: Addressed in commit `eb074aa55`: added the comma in the TECH_SPEC sentence for readability.
